lib/compiler/stack.js

```diff
@@ -0,0 +1,179 @@
+"use strict";
+
+class Stack {
+  constructor(ruleName, varName, type) {
```

Comment:
constructor docs.  for example, it wasn't clear that "type" could be "let".

Reply:
Added